src/system/sargon_os/sargon_os_factors.rs

```diff
@@ -21,11 +178,126 @@ impl SargonOS {
         self.load_private_device_factor_source(&device_factor_source)
             .await
     }
+}
 
-    /// Returns the "main Babylon" `DeviceFactorSource` of the current account as
-    /// a `DeviceFactorSource`.
-    pub fn bdfs(&self) -> DeviceFactorSource {
-        self.profile_holder.access_profile_with(|p| p.bdfs())
+impl SargonOS {
+    /// Adds all factor sources to Profile.
+    ///
+    /// # Emits Event
+    /// Emits `Event::ProfileModified { change: EventProfileModified::FactorSourceUpdated }`,
+    /// if any of the newly added FactorSources has **main** flag, then we remove the
+    /// main flag from the old BDFS.
+    ///
+    /// And also emits `Event::ProfileSaved` after having successfully written the JSON
+    /// of the active profile to secure storage.
+    async fn add_factor_sources_without_emitting_factor_sources_added(
```

Comment:
Why such notification shouldn't be emitted here?

Reply:
Because this is a "lower level" function which is used by two other higher level functions:
* `add_factor_source`
* `add_factor_sources` (note the plural)
And for these we wanna emit two different Notifications: `EventProfileModified::FactorSourceAdded` and `EventProfileModified::FactorSourcesAdded` (note the plural) respectively.